sdk/datafactory/Microsoft.Azure.Management.DataFactory/src/Generated/Models/WebActivityAuthentication.cs

```diff
@@ -56,36 +63,48 @@ public WebActivityAuthentication(string type, SecretBase pfx = default(SecretBas
 
         /// <summary>
         /// Gets or sets web activity authentication
-        /// (Basic/ClientCertificate/MSI)
+        /// (Basic/ClientCertificate/MSI/ServicePrincipal)
         /// </summary>
         [JsonProperty(PropertyName = "type")]
         public string Type { get; set; }
 
         /// <summary>
-        /// Gets or sets base64-encoded contents of a PFX file.
+        /// Gets or sets base64-encoded contents of a PFX file or Certificate
+        /// when used for ServicePrincipal
         /// </summary>
         [JsonProperty(PropertyName = "pfx")]
         public SecretBase Pfx { get; set; }
 
         /// <summary>
         /// Gets or sets web activity authentication user name for basic
-        /// authentication.
+        /// authentication or ClientID when used for ServicePrincipal. Type:
+        /// string (or Expression with resultType string).
         /// </summary>
         [JsonProperty(PropertyName = "username")]
-        public string Username { get; set; }
+        public object Username { get; set; }
```

Comment:
Is changing Username from `string` to `object` intentional?

Reply:
yes, object will accept other types, like expression, this is a requirement in DataFactory.

Reply:
Thank you.
Just for further understanding, what types or values are expected to be passed as a parameter for the `Username`?